### DISCUSSION TITLE
Modes for transition matrix calculation.

### DISCUSSION THREAD

**Discussion by EmilyMiddendorp** [theislab/cellrank#756] [Method/Algorithm]
Is there documentation on the modes available to calculate the transitions matrix? 
I am trying to decide which mode to use to analyse a dataset where we expect to find plastic event (dedifferentiation). So at the end I am interested on the absorption probabilities between the (biologically known) terminal states towards the initial state.

**Comment by michalk8** [comment]
> I am trying to decide which mode to use to analyse a dataset where we expect to find plastic event (dedifferentiation).

I am assuming you're referring to the `VelocityKernel`. The modes are explained in https://cellrank.readthedocs.io/en/stable/api/cellrank.tl.kernels.VelocityKernel.compute_transition_matrix.html#cellrank.tl.kernels.VelocityKernel.compute_transition_matrix
Briefly:
- `deterministic` - similarity between RNA velocity and connectivities (based on `scheme`, default is correlation, can be cosine, ...)
- `sampling` - sample 1 vector from normal distribution with velocity mean/var
- `monte_carlo` - same as above, but with many samples that are averaged
- `stochastic` - 2nd order approximation, see eq. 30 in our [preprint](https://www.biorxiv.org/content/10.1101/2020.10.19.345983v2.full.pdf)

For the backward velocities (relevant to your use case), only `deterministic` mode is usable and the options are:
- `negate` - instead of calculating the similarity of velocity vector `v` with connectivities `c`, calculates `sim(-v, c)`
- `transpose` - instead of calculating transition from a cell to its neighbors, do it vice-versa

There are also other approaches that don't require RNA velocity:
- if you have meaningful pseudotime values, you can also try inverting `PseudotimeKernel/CytoTRACEKernel` as:
```python
pk = ~PseudotimeKernel(adata, time_key=...)
pk.compute_transition_matrix(...)
```
- or if you have time-series info (e.g. days), I'd recommend taking look at `WOTKernel`, here's the [tutorial](https://cellrank.readthedocs.io/en/stable/real_time.html)

Lastly, if you're using the `GPCCA` estimator in the downstream analysis, sometimes, the macrostates can also correspond to initial states (in the forward case; this can bee seen in our CytoTRACE tutorial, cell 17: https://cellrank.readthedocs.io/en/stable/beyond_rna_velocity.html).
The GPCCA estimator has 2 private methods that you can use to export these to the `AnnData` object, if needed:
```python
GPCCA._compute_initial_states
GPCCA._set_initial_states_from_macrostates
```
